packages/rx_bloc_cli/mason_templates/bricks/lib_pin_code/__brick__/lib/lib_pin_code/services/app_pin_code_service.dart

```diff
@@ -0,0 +1,215 @@
+{{> licence.dart }}
+
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:widget_toolkit_pin/widget_toolkit_pin.dart';
+import '../repository/pin_code_repository.dart';
+
+class AppPinCodeService implements PinCodeService {
+  AppPinCodeService(
+    this._pinCodeRepository,
+  );
+
+  final PinCodeRepository _pinCodeRepository;
+
+  static const _isForFirstTime = 'isForFirstTime';
+  static const _firstPin = 'firstPin';
+  static const _secondPin = 'secondPin';
+  static const _storedPin = 'storedPin';
+  static const _isAuthenticatedWithBiometrics = 'isAuthenticatedWithBiometrics';
+  static const _areBiometricsEnabled = 'areBiometricsEnabled';
+  static const _areBiometricsEnabledWhileUsingTheApp =
+      'areBiometricsEnabledWhileUsingTheApp';
+  bool _isForFirstTimeExecuted = false;
+  bool _isVerificationPinProcess = true;
+  bool _isChangePin = false;
+  bool _isFromSessionTimeout = false;
+  bool? _isVerificationPinCorrect;
+
+  Future<void> deleteStoredPin() async {
+    await _pinCodeRepository.writePinToStorage(_storedPin, null);
+    await _pinCodeRepository.writePinToStorage(_firstPin, null);
+    await _pinCodeRepository.writePinToStorage(_secondPin, null);
+
+    await _pinCodeRepository.setBoolValue(_isForFirstTime, true);
+    await _pinCodeRepository.setBoolValue(
+        _isAuthenticatedWithBiometrics, false);
+    _isForFirstTimeExecuted = false;
+    _isVerificationPinCorrect = null;
+    _isVerificationPinProcess = true;
+    _isChangePin = false;
+  }
+
+  Future<void> deleteSavedData() async {
+    final enabledFromBefore = await _pinCodeRepository
+        .getBoolValue(_areBiometricsEnabledWhileUsingTheApp);
+    final current =
+        await _pinCodeRepository.getBoolValue(_areBiometricsEnabled);
+
+    if (current != null) {
+      if ((enabledFromBefore == true && current != true)) {
```

Comment:
Is there a reason to write it like this? Can you use here `if(enabledFromBefore && !current)`?

Reply:
This file was deleted and the logic was changed